sklearn/naive_bayes.py

```diff
@@ -117,6 +119,320 @@ def predict_proba(self, X):
+        """Get parameters for this metaestimator"""
+        return self._get_params('_models', deep=deep)
+
+    def set_params(self, **kwargs):
+        self._set_params('_models', **kwargs)
+        return self
+
+    def _validate_models(self, X):
+
+        valid_modules = copy.copy(__all__)
+        valid_modules.remove("GeneralNB")
+
+        self._cols = []
+        dict_col2model = {}
+        all_fit_priors = []
+        all_class_priors = []
+
+        if not isinstance(self.models, list):
+            raise TypeError(
+                "Expected list but got {}".format(type(self.models)))
+
+        names, _, _ = zip(*self.models)
+        self._validate_names(names)
+
+        for model in self.models:
+
+            # Check type of each entry in list
+            if not isinstance(model, tuple):
+                raise TypeError(
+                    "Expected list of tuples "
+                    "but got list of {}".format(type(model)))
+            if len(model) != 3:
+                raise ValueError("Expected tuple to have length of 3 "
+                                 "but got {}".format(len(model)))
+
+            _, estimator, cols = model
+
+            # Check naive bayes estimator for format
+            # `fit` and `_joint_log_likelihood` attributes
+            if callable(estimator):
+                raise ValueError("Estimator should be a callable specified.")
+            if not (hasattr(estimator, "fit")
+                    or hasattr(estimator, "_joint_log_likelihood")):
+                raise TypeError("Naive bayes estimator should implement "
+                                "the fit and _joint_log_likelihood methods. "
+                                "{} doesn't.".format(type(estimator)))
+            if estimator.__class__.__name__ not in valid_modules:
+                raise ValueError(
+                    "Distributions should be one of {}".format(valid_modules))
+
+            # Check naive bayes estimator for attributes
+            # like `priors`, `class_prior` and `fit_prior`
+            class_prior = getattr(estimator, "priors", None) or \
+                getattr(estimator, "class_prior", None)
+            fit_prior = getattr(estimator, "fit_prior", True)
+            all_class_priors.append(class_prior)
+            all_fit_priors.append(fit_prior)
+
+            # Check the columns for duplicate models and
+            # convert to feature if callable
+            if callable(cols):
+                cols = cols(X)
+            for col in cols:
+                if col in dict_col2model:
+                    raise ValueError("Duplicate specification of "
+                                     f"column {col} found.")
+                else:
+                    dict_col2model[col] = estimator.__class__.__name__.lower()
+            self._cols.append(cols)
+
+        # Check if class priors are the same throughout all estimators
+        if not all(prior is None for prior in all_class_priors):
```

Comment:
I think this case may be rare enough that we can rely on checking the learnt priors, no?

Reply:
Removed this part because checking of learnt priors is now removed.